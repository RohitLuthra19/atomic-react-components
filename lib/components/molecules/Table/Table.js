// @flow
/**
 *
 * Table
 *
 */
import React from 'react';
import styled from 'styled-components';

import styles from './Table.style';
import type { Props } from './types';
import Input from '../InputChoice';
import Button from '../../atoms/Button';

const Table = ({ data, isCheckedBox, isSubmitButton }: Props): Node => {
  const columnName = Object.keys(data[0]);

  const headers = (
    <tr>
      {isCheckedBox && <th key="checkBoxHead">select </th>}
      {columnName.map((name, n) => (
        // eslint-disable-next-line react/no-array-index-key
        <th key={n}>{name}</th>
      ))}
      {isSubmitButton && <th> Action </th>}
    </tr>
  );

  const rows = data.map((item, i) => (
    // eslint-disable-next-line react/no-array-index-key
    <tr key={i}>
      {isCheckedBox && (
        <td>
          <Input
            type="checkbox"
            aria-label="row select"
            id={`rowSelect${i}`}
            name={`rowSelect${i}`}
          />
        </td>
      )}
      {columnName.map((prop, p) => (
        // eslint-disable-next-line react/no-array-index-key
        <td key={p}>{item[prop]}</td>
      ))}
      {isSubmitButton && (
        <td>
          <Button type="button" aria-label="submit">
            Submit
          </Button>
        </td>
      )}
    </tr>
  ));

  return (
    <table>
      <thead>{headers}</thead>
      <tbody>{rows}</tbody>
    </table>
  );
};

export default styled(Table)`
  ${styles};
`;

export { Table as TableVanilla };
