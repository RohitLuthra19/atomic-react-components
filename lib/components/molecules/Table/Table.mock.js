const defaultConfig = {
  data: [
    { ClientName: 'US Bank', Payer: 'test', value: 'v1' },
    { ClientName: 'Test Client', Payer: 'test2', value: 'v2' },
  ],
};

export default defaultConfig;
