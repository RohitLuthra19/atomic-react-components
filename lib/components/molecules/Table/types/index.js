// @flow
import type { Node } from 'react';

export type Props = {
  className?: string,
  data: Array<mixed>,
  children: Node,
  vertical?: boolean,
  dataMappingKey: mixed,
};

export type HeadProps = {
  children: node,
};
