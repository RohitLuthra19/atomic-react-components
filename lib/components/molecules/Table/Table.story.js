import React from 'react';
import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import defaultConfig from './Table.mock';

// Import Styled Component to showcase variations
import Table, { TableVanilla } from '.';

// TODO: indentation is not working fine in editor. Need to check for .hbs files
// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module)
  .addParameters({ jest: ['Table', 'TableVanilla'] })
  .add('CTA Knobs', () => <TableVanilla {...defaultConfig} isSubmitButton isCheckedBox />)
  .add('CTA', () => <Table {...defaultConfig} isCheckedBox />);
