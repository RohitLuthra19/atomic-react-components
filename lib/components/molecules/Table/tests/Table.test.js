import { shallow } from 'enzyme';

import Table from '../index';
import defaultConfig from '../Table.mock';

describe('<Table />', () => {
  const TableComponent = shallow(<Table {...defaultConfig}>Test</Table>);
  const table = TableComponent.find('table');
  const thead = table.find('thead');
  const tbody = table.find('tbody');

  it('should have table tag', () => {
    expect(TableComponent.find('table')).toBeTruthy();
  });

  it('should render columns based on thead', () => {
    expect(thead).toBeTruthy();
  });

  it('should render rows based on tbody', () => {
    expect(tbody).toBeTruthy();
  });

  test('should render correctly', () => {
    expect(TableComponent).toMatchSnapshot();
  });
});
