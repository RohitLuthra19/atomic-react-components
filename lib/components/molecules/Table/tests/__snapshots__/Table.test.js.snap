// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render correctly 1`] = `
<StyledComponent
  data={
    Array [
      Object {
        "ClientName": "US Bank",
        "Payer": "test",
        "value": "v1",
      },
      Object {
        "ClientName": "Test Client",
        "Payer": "test2",
        "value": "v2",
      },
    ]
  }
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Table-sc-1ho1shi-0",
        "isStatic": false,
        "rules": Array [
          "border-spacing:0;border:1px solid black;tr{:last-child{td{border-bottom:0;}}}th,td{margin:0;padding:0.5rem;border-bottom:1px solid black;border-right:1px solid black;:last-child{border-right:0;}}",
          [Function],
          ";",
          ";",
        ],
      },
      "displayName": "Table",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Table-sc-1ho1shi-0",
      "target": [Function],
      "toString": [Function],
      "usesTheme": false,
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Test
</StyledComponent>
`;
